src/main/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporter.java

```diff
@@ -0,0 +1,355 @@
+
+            final MultiMap headers = MultiMap.caseInsensitiveMultiMap()
+                    .add(HttpUtil.ACCEPT_HEADER, HttpHeaderValues.APPLICATION_JSON)
+                    .add(HttpUtil.CONTENT_TYPE_HEADER, HttpHeaderValues.APPLICATION_JSON);
+
+            final Future<HttpClientResponse> responseFuture = httpClient.post(
+                    greenbidsAnalyticsProperties.getAnalyticsServer(),
+                    headers,
+                    commonMessageJson,
+                    greenbidsAnalyticsProperties.getTimeoutMs());
+
+            return responseFuture.compose(response -> {
+                final int responseStatusCode = response.getStatusCode();
+                if (responseStatusCode == 202 || responseStatusCode == 200) {
+                    return Future.succeededFuture();
+                } else {
+                    return Future.failedFuture(
+                            "Unexpected response status: " + response.getStatusCode());
+                }
+            });
+
+        } catch (EncodeException e) {
+            return Future.failedFuture("Failed to encode as JSON: " + e.getMessage());
+        }
+
+    }
+
+    private Boolean isSampled(double samplingRate, String greenbidsId) {
+        if (samplingRate < 0 || samplingRate > 1) {
+            logger.warn("Warning: Sampling rate must be between 0 and 1");
+        }
+
+        final double exploratorySamplingRate = samplingRate
+                * greenbidsAnalyticsProperties.getExploratorySamplingSplit();
+        final double throttledSamplingRate = samplingRate
+                * (1.0 - greenbidsAnalyticsProperties.getExploratorySamplingSplit());
+
+        final long hashInt = Integer.parseInt(
+                greenbidsId.substring(greenbidsId.length() - 4), 16);
+        final boolean isPrimarySampled = hashInt < exploratorySamplingRate * RANGE_16_BIT_INTEGER_DIVISION_BASIS;
+
+        return isPrimarySampled
+                || hashInt >= (1 - throttledSamplingRate) * RANGE_16_BIT_INTEGER_DIVISION_BASIS;
+    }
+
+    private CommonMessage createBidMessage(
+            AuctionContext auctionContext,
+            BidResponse bidResponse,
+            String greenbidsId,
+            String billingId) {
+        final List<Imp> imps = Optional.ofNullable(auctionContext)
+                .map(AuctionContext::getBidRequest)
+                .map(BidRequest::getImp)
+                .orElse(Collections.emptyList());
+
+        if (CollectionUtils.isEmpty(imps)) {
+            throw new IllegalArgumentException("AdUnits list should not be empty");
+        }
+
+        final long auctionElapsed = Optional.ofNullable(auctionContext.getBidRequest())
+                .map(BidRequest::getExt)
+                .map(ExtRequest::getPrebid)
+                .map(ExtRequestPrebid::getAuctiontimestamp)
+                .map(timestamp -> System.currentTimeMillis() - timestamp).orElse(0L);
```

Comment:
for the time tracking we usually use the `Clock clock` dependency, it's helpful also because it can be mocked in the unit tests